sdk/Pulumi/Deployment/Deployment_Run.cs

```diff
@@ -341,6 +332,15 @@ internal static async Task<int> CreateRunnerAndRunAsync(
             Func<Deployment> deploymentFactory,
             Func<IRunner, Task<int>> runAsync)
         {
+            if (Environment.GetEnvironmentVariable("PULUMI_ATTACH_DEBUGGER") == "true")
+            {
+                while (!System.Diagnostics.Debugger.IsAttached)
+                {
+                    // keep waiting until the debugger is attached
+                    await Task.Delay(1000).ConfigureAwait(false);
```

Comment:
Nice change from `Thread.Sleep(1)`: both converting it to async and waiting longer than 1 ms. I don't think 1s is noticeable that much, but should we go for 100 ms instead to make sure the experience is as best as possible?

Reply:
I'd tried 500ms and 1000ms and didn't see any noticeable differences in the experience from my machine, but went ahead and reduced to 100ms to make sure things are fast when attaching.